apps/snitch_core/lib/core/data/model/user.ex

```diff
@@ -23,7 +23,9 @@ defmodule Snitch.Data.Model.User do
 
   @spec get(map | non_neg_integer) :: UserSchema.t() | nil
   def get(query_fields_or_primary_key) do
-    QH.get(UserSchema, query_fields_or_primary_key, Repo)
+    UserSchema
+    |> QH.get(query_fields_or_primary_key, Repo)
+    |> Repo.preload(:role)
```

Comment:
didn't want to pollute controllers with any Repo word so thought of doing this, but it will happen eventually so no use, will move this to the controller then.